assets/schema_input.json

```
@@ -12,25 +12,67 @@
                 "pattern": "^\\S+$",
                 "errorMessage": "Sample name must be provided and cannot contain spaces"
             },
-            "fastq_1": {
-                "type": "string",
-                "pattern": "^\\S+\\.f(ast)?q\\.gz$",
-                "errorMessage": "FastQ file for reads 1 must be provided, cannot contain spaces and must have extension '.fq.gz' or '.fastq.gz'"
+            "bam": {
+                "errorMessage": "BAM file cannot contain spaces and must have extension '.bam'",
+                "anyOf": [
+                    {
+                        "type": "string",
+                        "pattern": "^\\S+\\.bam$"
+                    },
+                    {
+                        "type": "string",
+                        "maxLength": 0
+                    }
+                ],
```

```suggestion
                "type": "string",
                "pattern": "^\\S+\\.bam$",
```

You don't need to specify empty fields in the schema, `nf-validation` automatically skips validation of empty fields, if they are not required :). And I think `anyOf` doesn't work for the conversion to a channel (I've not tested it though).

---

I copy paste code from another PR, if it's ok with you, I'd rather fix this part in a dedicated PR

---

Yeah that's okay! :)